### ISSUE TITLE
assoc types: bootstrapping rust generates a rustdoc error

### ISSUE THREAD

**Issue by japaric** [rust-lang/rust#20402]
On my PRs (#20390 and #20398) that add associated types to stdlib traits, I have observed that bootstrapping rust generates the following rustdoc error:

```
$ ../configure && make
...
rustdoc: doc/not_found.html
rustdoc: doc/index.html
rustdoc: doc/intro.html
rustdoc: doc/tutorial.html
rustdoc: doc/guide.html
rustdoc: doc/guide-ffi.html
rustdoc: doc/guide-macros.html
rustdoc: doc/guide-ownership.html
rustdoc: doc/guide-tasks.html
rustdoc: doc/guide-container.html
rustdoc: doc/guide-pointers.html
rustdoc: doc/guide-testing.html
rustdoc: doc/guide-plugin.html
rustdoc: doc/guide-crates.html
rustdoc: doc/complement-bugreport.html
rustdoc: doc/guide-error-handling.html
rustdoc: doc/complement-lang-faq.html
rustdoc: doc/complement-design-faq.html
rustdoc: doc/complement-project-faq.html
rustdoc: doc/rustdoc.html
rustdoc: doc/guide-unsafe.html
rustdoc: doc/guide-strings.html
cp: doc/full-toc.inc
rustdoc: doc/reference.html
rustdoc: doc/libc/index.html
rustdoc: doc/std/index.html
thread '<unnamed>' panicked at 'can't make docs for trait item with name Some(Item)', /root/rust/src/librustdoc/html/render.rs:2125
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: ()', /root/rust/src/libcore/result.rs:746
make: *** [doc/std/index.html] Error 101
```

Rennuning `make` yields the same error. **But** running `make -jN` (with `N > 2`) successfully completes the process ...

If this happens on bors, then this would block landing those two PRs.

cc @aturon @nikomatsakis

**Comment by aturon** [comment]
cc @alexcrichton

**Comment by aturon** [comment]
cc @cmr @tomjakubowski

**Comment by japaric** [comment]
This was fixed in #20410, thanks @cmr!
